### ISSUE TITLE
Extract features return different layer values (vectors) each time, is it working well?

### ISSUE THREAD

**Issue by hairzooc** [google-research/bert#312]
Hey guys,
I've tested extract_features.py with same input text and found that it returns different layer values each time.
(From source: layer_value = [round(float(x), 6) for x in layer_output[i:(i + 1)].flat]

I just expected if I enter same input then I would get same output. But outputs are not same.
Is it normal? If you guys have some idea on that, please let me know.
Thanks in advance :)

**Comment by hairzooc** [comment]
For your understanding, I'm sharing the output vector values for 'hello world'

1) first time
[CLS] [0.476536, 0.372158, 1.465642]
hello [0.111987, 1.128579, 0.990834]
world [0.416854, 1.073154, 1.456178]
[SEP] [-0.338621, 0.568671, 0.049002]

2) second time
[CLS] [0.842703, 0.942707, 0.420058]
hello [0.866664, 0.40403, -0.005693]
world [1.921718, -0.325165, -1.704712]
[SEP] [-0.475752, 0.405486, -1.818929]

As you can see, even if I input same sentence 'hello world', output vector is not same.
I'd like to know how I can get fixed output vector for same sentence if possible.
I hope you guys can give me an advice on that.
Thank you!

**Comment by arcral** [comment]
I have the same question. Same inputs, but difference outouts each time. I'mconfused about it. Have you solved it? Or is it normal?

**Comment by arcral** [comment]
@hairzooc it's caused by dropout.

**Comment by Liujish931** [comment]
@arcral  is right, you have to zero dropout in evaluation mode: bert_model.eval() or you can manually do it by setting the dropout attributes in each layer

**Comment by hairzooc** [comment]
I have found that it is caused by do_lower_case = True
For multilingual model, this option should be False.
Thank you for your help :)
